From babc121e36ec93ceb4c09a0157c75deaab98691a Mon Sep 17 00:00:00 2001
From: Sudheesh Mavila <sudheesh.mavila@amd.com>
Date: Tue, 10 Nov 2020 12:42:28 +0530
Subject: [PATCH 34/48] amd-xgbe: rrc is required only for Fixed PHY
 configuration

xgbe driver does RRC in every 10 seconds if the link is down.
When AN is enabled and KR training is in progress,
doing RRC cause KR training failure. The patch enables RRC only
if AN is disabled.

Signed-off-by: Sudheesh Mavila <sudheesh.mavila@amd.com>
---
 drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c b/drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c
index 371b4c91e289..73f8bf084021 100644
--- a/drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c
+++ b/drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c
@@ -2656,7 +2656,8 @@ static int xgbe_phy_link_status(struct xgbe_prv_data *pdata, int *an_restart)
 	/* No link, attempt a receiver reset cycle */
 	if (phy_data->rrc_count++ > XGBE_RRC_FREQUENCY) {
 		phy_data->rrc_count = 0;
-		xgbe_phy_rrc(pdata);
+		if (pdata->phy.autoneg == AUTONEG_DISABLE)
+			xgbe_phy_rrc(pdata);
 	}
 
 	return 0;
-- 
2.27.0

